docs/adr/0002-deprecate-kubebuilder-defaults.md

```diff
@@ -0,0 +1,52 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Deprecate kubebuilder defaults](#deprecate-kubebuilder-defaults)
+  - [Context](#context)
+  - [Decision](#decision)
+  - [Consequences](#consequences)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Deprecate kubebuilder defaults
+<!-- A short and clear title which is prefixed with the ADR number -->
+
+<!-- TODO: remove the following disable link checker comment before committing your ADR -->
+<!-- markdown-link-check-disable-next-line -->
+- Status: accepted
+- Date: 2024-05-21
+- Authors: @alexander-demicev
+- Deciders: @Danil-Grigorev @furkatgofurov7 @salasberryfin @mjura @yiannistri
+
+## Context
+<!-- What is the context of the decision and what's the motivation -->
+
+Enforcing default values for fields in CRD definitions can cause problem with the GitOps approach. It can lead to a situation where defined resources in Git repository differs from the actual state of applied resources in the cluster. This can lead to confusion and unexpected behavior.
```

Comment:
```suggestion
Enforcing default values for fields in CRD definitions can cause problem with the GitOps approach. It can lead to a situation where defined resources in Git repository differs from the actual state of applied resources in the cluster. This can lead to unexpected behavior like state drift, unless is manually handled.
```